### ISSUE TITLE
Check for existence of main class in case it's already included in another plugin.

### ISSUE THREAD

**Issue by pderksen** [pantheon-systems/wp-native-php-sessions#81]
Some plugins would like to include the entire WP Native PHP Sessions plugin inside their own. This is the case with my plugin WP Simple Pay ([Lite](https://github.com/moonstonemedia/WP-Simple-Pay-Lite-for-Stripe) & Pro).

For the cases where a plugin including the `Pantheon_Sessions` class as well as the standalone WP Native PHP Sessions plugin are both activated, a simple 'class_exists' check should avoid any errors.

**Comment by danielbachhuber** [comment]
> Some plugins would like to include the entire WP Native PHP Sessions plugin inside their own.

Is there a reason you can't simply encourage the user to install this plugin?

I don't know that what you suggest is a use case we want to officially support:

1. Because of how WordPress loads plugins, the user may get your copy of WP Native PHP Sessions or the real plugin.
2. Any improvements we make, bugs we fix, etc. will need to be ported over to your plugin. There's the chance the user could be stuck with an older version of WP Native PHP Sessions because you haven't updated the copy in the plugin (or they've decided to stop paying for plugin updates, etc.).

**Comment by pderksen** [comment]
@danielbachhuber Up until now we've integrated the plugin [WP Session Manager](https://github.com/ericmann/wp-session-manager) into our plugin, following suit with plugins like [Give](https://github.com/WordImpress/Give/tree/release/1.8.18/includes/libraries/sessions), [EDD](https://github.com/easydigitaldownloads/easy-digital-downloads/tree/master/includes/libraries), and [Ninja Forms](https://github.com/wpninjas/ninja-forms/tree/master/includes/Libraries/Session) (links are to their integrations).

I was hoping that by including it with WP Simple Pay so it "just works" (regardless of how sessions are treated on their server) would be the best route. I'm also trying to reduce support on our end. I'm pretty sure many of our our new users will be contacting our support before installing WP Native PHP Sessions without checking our docs first. Not sure what's best for both of our plugins at this point though.

**Comment by danielbachhuber** [comment]
Makes sense, thanks for the additional detail.

At this point, I don't think it makes sense for WP Native PHP Sessions to formally support being embedded into other plugins.

**Comment by pderksen** [comment]
No sweat. I'll keep tabs on changes here either way. Thanks!

**Comment by pderksen** [comment]
@danielbachhuber Just an update. In our latest releases to WP Simple Pay Lite & Pro, we went back to integrating the latest version of WP Session Manager into our plugin (again like Give, EDD, and Ninja Forms), and Pantheon was the only host we tested that didn't work with sessions implemented this way.

We went ahead and added a constant that allows the site owner to force use of PHP sessions (instead of the default table storage method WP Sessions Manager uses).

Finally, we check for the existence of WP Native PHP Sessions (this plugin) and force use of PHP sessions if detected.

https://github.com/moonstonemedia/WP-Simple-Pay-Lite-for-Stripe/blob/75d4560d311871f7d80309888337550a56117822/includes/core/session.php#L290-L297
